python_modules/libraries/dagster-aws/dagster_aws/sfn/sfn_launcher.py

```diff
@@ -0,0 +1,161 @@
+DEFAULT_RUN_TASK_RETRIES = 3
+
+
+class SFNFinishedExecutioinError(Exception):
+    pass
+
+
+class SFNLauncher(RunLauncher[T_DagsterInstance], ConfigurableClass):
+    def __init__(
+        self,
+        inst_data: Optional["ConfigurableClassData"],
+        sfn_arn: str,
+        name: Optional[str] = None,
+        input: Optional[str] = None,  # noqa: A002
+        trace_header: Optional[str] = None,
+    ) -> None:
+        self._client: SFNClient = boto3.client("stepfunctions")
+        self._inst_data = inst_data
+        self._sfn_arn = check.str_param(sfn_arn, "sfn_arn")
+        self._name = check.opt_str_param(name, "name") if name is not None else None
+        self._input = check.opt_str_param(input, "input") if name is not None else None
+        self._trace_header = (
+            check.opt_str_param(trace_header, "trace_header") if trace_header is not None else None
+        )
+
+    @cached_property
+    def _start_execution_input(self) -> "StartExecutionInputRequestTypeDef":
+        input_dict: StartExecutionInputRequestTypeDef = {"stateMachineArn": self._sfn_arn}
+        if self._name is not None:
+            input_dict["name"] = self._name
+        if self._input is not None:
+            input_dict["input"] = self._input
+        if self._trace_header is not None:
+            input_dict["traceHeader"] = self._trace_header
+        return input_dict
+
+    def launch_run(self, context: LaunchRunContext) -> None:
+        try:
+            execution_arn = self._client.start_execution(**self._start_execution_input)[
+                "executionArn"
+            ]
+
+        except ClientError as err:
+            error_info = err.response.get("Error", {})
+            logging.error(
+                "Couldn't start execution %s. Here's why: %s: %s",
+                self._sfn_arn,
+                error_info.get("Code", "Unknown error"),
+                error_info.get("Message", "No error message available"),
+            )
+            raise
+
+        try:
+            response = self._wait_for_execution_completion(execution_arn)
+        except DagsterExecutionInterruptedError as e:
+            logging.error(f"Error waiting for execution completion: {e}")
+            _ = backoff(
+                self._stop_execution,
+                retry_on=(ClientError,),
+                kwargs={"execution_arn": execution_arn},
+                max_retries=int(
+                    os.getenv("RUN_TASK_RETRIES", DEFAULT_RUN_TASK_RETRIES),
+                ),
+            )
+            raise
+        else:
+            logging.info(
+                f"Execution {execution_arn} completed successfully, Status: {response['status']}"
+            )
+            if response["status"] != "SUCCEEDED":
+                raise SFNFinishedExecutioinError(
+                    f"Step Function execution {self._sfn_arn} run {execution_arn} completed with status {response['status']} :\n{response.get('errorMessage')}",
+                )
```

Comment:
* `launch_run` shouldn't block on the launched run process completing 
* `context` is never used anywhere so I dont see how this could be executing the appropriate target

Reply:
I initially started with pipes and copy-pasted the code for the launcher. However, I made improvements and restructured it to work with the ECS launcher.

The context for launching a run is now provided due to the inheritance behavior of the RunLauncher.